domain/schema/controller.go

```diff
@@ -591,3 +592,42 @@ CREATE TABLE flag (
 );
 `)
 }
+
+// The permissions schema
+func permissionsSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE access_type (
+    id     INT PRIMARY KEY,
+    type   Text
+);
+
+CREATE UNIQUE INDEX idx_access_type
+ON access_type (type);
+
+-- Maps to the Access type in core/permission package.
+INSERT INTO access_type VALUES
+    (0, ''),
+    (1, 'read'),
+    (2, 'write'),
+    (3, 'consume'),
+    (4, 'admin'),
+    (5, 'login'),
+    (6, 'add-model;'),
+    (7, 'superuser');
+
+CREATE TABLE permissions (
+    access_type_id  INT NOT NULL,
```

Comment:
Then there should be another table that has the pair (read permission + model) against a particular entity UUID.

That table then has a join table that links the specific permissions with a user.

Reply:
added.